problem2.py

```diff
@@ -0,0 +1,33 @@
+"""
+Time Complexity : O(N)
+Space Complexity :O(N)
```

Comment:
To be more concise,i t will be O(H) where h is the height of the tree which when given a skewed tree can become O(N) because N and H there will be same